features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/apis/v1/tags_api_router.jag

```diff
@@ -85,37 +86,67 @@ routeManager.register('PUT', 'publisher', '/publisher/api/tag/{type}/{id}', func
     var msg = {};
     msg['status'] = 200;
     msg['ok'] = true;
+    msg['message'] = null;
 
-    var artifactManager = rxtManager.getArtifactManager(type);
-
-    var artifact = artifactManager.get(id);
-
-    var model = modelManager.getModel(type);
-    model.import('asset', artifact);
+    var userManager = server.userManager(server.current(session).tenantId);
+    var user = server.current(session);
+    var permission = null;
 
-    var report = model.validate();
-
-    //Check if the model is valid, the important validation here is checking if the asset is in the published state
-    if ((report) && (report.failed)) {
-        response.sendError(404, 'Unable to attach tag');
-        //msg.ok=false;
-        //msg.status=404;
-        // print(msg);
+    if (type == 'webapp') {
```

Comment:
Formatting and commenting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files